sabnzbd/par2file.py

```diff
@@ -243,4 +243,7 @@ def parse_par2_file(fname: str, md5of16k: Dict[bytes, str]) -> Tuple[str, Dict[s
             old_name = md5of16k.pop(hash16k)
             logging.debug("Par2-16k signature of %s not unique, discarding", old_name)
 
+    # Sort table by filename
+    table = {filename: table[filename] for filename in sorted(table.keys())}
+
```

Comment:
In the end this line is only needed for the extremely specific case that 2 consecutive files have the same crc32?
Doesn't the other part of the change already fix it?

Reply:
Correct this part is to handle duplicates within the set.

Without the sorting we would get par2 packets in whatever order they are in the file; so in the issue it's 003, 004, 001, 002.

Without this change, we'd first check 003, loop through `nzf_list` and the first file match would be 002 so it would rename, then when we get to 002 it'd want to rename to 003 however on Linux/Mac `shutil.move` will overwrite so the end result is missing 003.

This would all be renaming the same file multiple times:
002 => 003 (overwrite) => 002

By combining both both lists are processed in filename order:

001 == 001 OK
002 == 002 OK
003 == 003 OK (skips nzf_list 002 because of other changes; path already found to be valid)
004 == 004 OK

Reply:
Could you add that to the comment why we do this?
Then I'll merge.

Reply:
Done, log of behaviour without the sorting showing the 002 to 003 to 002 behaviour.

```
2023-08-04 08:47:41,456::DEBUG::[newsunpack:1994] Quick-check will rename f0dN5qb5L0RpSL7E0dkrQS5SkWSUKiik.7z.002 to f0dN5qb5L0RpSL7E0dkrQS5SkWSUKiik.7z.003
2023-08-04 08:47:41,457::DEBUG::[newsunpack:1980] Quick-check of file f0dN5qb5L0RpSL7E0dkrQS5SkWSUKiik.7z.004 OK
2023-08-04 08:47:41,457::DEBUG::[newsunpack:1980] Quick-check of file f0dN5qb5L0RpSL7E0dkrQS5SkWSUKiik.7z.001 OK
2023-08-04 08:47:41,457::DEBUG::[newsunpack:1994] Quick-check will rename f0dN5qb5L0RpSL7E0dkrQS5SkWSUKiik.7z.003 to f0dN5qb5L0RpSL7E0dkrQS5SkWSUKiik.7z.002
```